spec/views/idv/usps/index.html.erb_spec.rb

```diff
@@ -12,7 +12,7 @@
 
     expect(usps_presenter).to receive(:title)
     expect(usps_presenter).to receive(:button)
-    expect(usps_presenter).to receive(:cancel_path)
+    expect(usps_presenter).to receive(:back_path).and_return('/')
```

Comment:
Ok so this is clearly not changed here, but this is a really weird way to stub IMO, I would just add this as an attribute of the instance double above...


```
usps_presenter = instance_double(Idv::UspsPresenter, back_path: '/')
```

Also again, not changed here but I'm pretty sure we don't even need the `allow(Idv::UspsPresenter).to receive(:new)` if we're just going to directly assign `@presenter` here

I'd love it if we could clean this up while we're here, but if not LMK and I can make a note to follow up after this PR is merged

Reply:
Happy to make changes here. I wasn't aware of the `instance_double` overload to pass stubbed values. I like that better than what I'd proposed.

The comment has me thinking though... what exactly are we testing here? That the view calls methods on the presenter? I'm not sure I find that especially valuable as a test.

I'd kinda wonder if it makes more sense to assert things about the view, given a real presenter instance. Things like.. it has a ["Send a letter"|"Send another letter"] if the user ["hasn't"|"has"] requested one, it has a Back button that links to ["phone verify"|"OTP entry"] if user is ["mid-verify"|"returning to verify"].

Do you think that'd make sense? Or is there a more appropriate place we'd be testing those things?

Reply:
I have a feeling these were all added as a way to stub out a blank/null object

I agree asserting on the results (view contains XYZ) is a clearer pattern for expectations, and the specs will blow up if the right methods/fields aren't stubbed out.

Reply:
I pushed up some changes in 399a9da to try to improve the specs for this view. It might not be exactly how we'd envisioned in that it's not really enforce stubbing all the methods of a presenter double, but I think as implemented it feels to me to give good coverage for all the branches of the presenter's effect on view logic. Definitely open to any feedback though.

Reply:
Yeah I think the changes are definitely an improvement! In an idea world we'd use an entirely real or entirely fake object to minimize what we stub, but this is close enough for now.

Reply:
Aside: After tinkering more with other tests in recent changes, I think I understand what you're saying. The changes in 399a9da could be arguably _too_ real as far as trying to anticipate a bunch of real-world underlying conditions for why the view would render in a certain way. I think those might make most sense as part of the presenter spec, where the view spec would more plainly take properties from the presenter, and not care how those properties come to be assigned.